core/state_update.go

```diff
@@ -199,3 +217,14 @@ func noncesDigest(nonces map[felt.Felt]*felt.Felt, digest *crypto.PoseidonDigest
 		digest.Update(&addr, nonces[addr])
 	}
 }
+
+func EmptyStateDiff() *StateDiff {
+	return &StateDiff{
+		StorageDiffs:      make(map[felt.Felt]map[felt.Felt]*felt.Felt),
+		Nonces:            make(map[felt.Felt]*felt.Felt),
+		DeployedContracts: make(map[felt.Felt]*felt.Felt),
+		DeclaredV0Classes: make([]*felt.Felt, 0),
+		DeclaredV1Classes: make(map[felt.Felt]*felt.Felt),
+		ReplacedClasses:   make(map[felt.Felt]*felt.Felt),
+	}
```

Comment:
This is what I was mentioning in the comment above about StorageDiff. The same way we could consider using type alias like Address or ClassHash for example, and so DeployedContracts map would be: `map[Address]ClassHash`.
 This allows the code to be more readable, you could also make it safer as you restrict the specific type that must be passed to a function (not just any felt), and you could even create methods on top of these structures. Wdyt?

Reply:
This makes sense to me too. Can this text be part of the issue description 🙏 with extra details if possible

Reply:
This is also added in issue #2396